svthalia/concrexit
# Optional event registration tweaks

Closes #1771

### Summary
#1574 enabled a new feature for a lot of events. However, it also 'reopened' event registration for events in the past. This PR fixes that, and makes this whole feature optional, so event organizers can also disable it.

### How to test
1. Have an event
2. You cannot register
3. Make optional registrations allowed
4. You can register
5. Check out other edge cases
